### PR TITLE
Fix DRM mode setting to use SFML window dimensions

### PR DESCRIPTION
**Pull request by mickelson**
## Description

This PR merges the change from #3136 into master

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
Please rebase on `master` to get rid of that merge commit

**Comment by ChrisThrasher** [comment]
Rebased on `master`, fixed formatting issues, and fixed clang-tidy errors.

File: src/SFML/Window/Unix/JoystickImpl.cpp [hunk 1]
```diff
@@ -554,9 +554,7 @@ JoystickCaps JoystickImpl::getCapabilities() const
     // Get the number of buttons
     char buttonCount = 0;
     ioctl(m_file, JSIOCGBUTTONS, &buttonCount);
-    caps.buttonCount = static_cast<unsigned int>(buttonCount);
-    if (caps.buttonCount > Joystick::ButtonCount)
-        caps.buttonCount = Joystick::ButtonCount;
+    caps.buttonCount = std::min(static_cast<unsigned int>(buttonCount), Joystick::ButtonCount);
```
Comment (kimci86) [inline src/SFML/Window/Unix/JoystickImpl.cpp:557]: Unrelated change.

Comment (ChrisThrasher) [reply src/SFML/Window/Unix/JoystickImpl.cpp:557]: It's fine. I made this change to fix a clang-tidy warning.

Comment (ChrisThrasher) [reply src/SFML/Window/Unix/JoystickImpl.cpp]: Ah I was using a source build of clang-tidy in my Ubuntu 24 machine which meant I had access to checks that don't exist in clang-tidy-18, the version of clang-tidy we use to analyze all the Linux code, hence me seeing a local clang-tidy failure that didn't exist in CI. I reverted this. It's worth fixing but this PR is not the place for that.

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/DRM/VideoModeImpl.cpp [hunk 2]
```diff
@@ -53,7 +53,12 @@ std::vector<VideoMode> VideoModeImpl::getFullscreenModes()
 ////////////////////////////////////////////////////////////
 VideoMode VideoModeImpl::getDesktopMode()
 {
-    if (const auto* ptr = DRMContext::getDRM().mode)
+    const Drm&         drm = DRMContext::getDRM();
+    drmModeModeInfoPtr ptr = drm.mode;
+    if (!ptr) // if no mode has been set, return the original mode we started with
+        ptr = &(drm.originalCrtc->mode);
+
+    if (ptr)
```
Comment (ChrisThrasher) [inline src/SFML/Window/DRM/VideoModeImpl.cpp:61]: At this point it's not possible for `ptr` to be null. `&(drm.originalCrtc->mode)` has to return a non-null address. You can remove this conditional.

File: src/SFML/Window/DRM/VideoModeImpl.cpp [hunk 3]
```diff
@@ -53,7 +53,12 @@ std::vector<VideoMode> VideoModeImpl::getFullscreenModes()
 ////////////////////////////////////////////////////////////
 VideoMode VideoModeImpl::getDesktopMode()
 {
-    if (const auto* ptr = DRMContext::getDRM().mode)
+    const Drm&         drm = DRMContext::getDRM();
+    drmModeModeInfoPtr ptr = drm.mode;
```
Comment (ChrisThrasher) [inline src/SFML/Window/DRM/VideoModeImpl.cpp:57]:
```suggestion
    drmModeModeInfoPtr ptr = DRMContext::getDRM();
```

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

**Review by ChrisThrasher — APPROVED** [review]
@mickelson I've made some changes to this PR to fix CI errors and address some other cleanliness and style matters.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.
